### PR TITLE
feat:API compleation -Sorting and searching (#40313)

### PR DESCRIPTION
**Pull request by nx-mukesh**
On trainee API with swagger Pending to implement.

### CODE REVIEW

File: src/libs/routes/authMiddleWare.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ const authMiddleware = (module, permissionType) => async (req, res, next) => {
   console.log('inAuthMiddle', { userData });
   if (!userData) {
     next({
-      err: 'Unauthorized',
+      err: 'Unauthorized(3)',
```
Comment (namanpar) [inline src/libs/routes/authMiddleWare.ts]: key must error and do not extra characters like (1).
Use proper status code or message
see http status code

File: src/libs/routes/authMiddleWare.ts [hunk 2]
```diff
@@ -16,15 +16,15 @@ const authMiddleware = (module, permissionType) => async (req, res, next) => {
...
+      err: 'Unauthorized (1)',
       message: 'User not Authorized to access..!!',
       status: 403,
     });
   }
 
   if (!user) {
     next({
-      error: 'unauthorized',
+      error: 'unauthorized (2)',
```
Comment (namanpar) [inline src/libs/routes/authMiddleWare.ts]: key must error and do not extra characters like (1).
Use proper status code or message
see http status code

File: src/libs/routes/authMiddleWare.ts [hunk 3]
```diff
@@ -16,15 +16,15 @@ const authMiddleware = (module, permissionType) => async (req, res, next) => {
     user = jwt.verify(token, secret);
   } catch (err) {
     next({
-      err: 'Unauthorized',
+      err: 'Unauthorized (1)',
```
Comment (namanpar) [inline src/libs/routes/authMiddleWare.ts]: key must error and do not extra characters like (1).
Use proper status code or message
see http status code

File: src/repositories/user/UserRepository.ts [hunk 4]
```diff
@@ -4,40 +4,40 @@ import IUserModel from './IUserModel';
...
+	public countData(): Query<number, IUserModel> {
+		return super.count();
+	}
+
+	public async create(data: any): Promise<IUserModel> {
+		return super.create(data);
+	}
+
+	public delete(data): UpdateQuery<IUserModel> {
+		return super.softDelete({ originalId: data.originalId, deleteAt: undefined }, data.originalId);
```
Comment (namanpar) [inline src/repositories/user/UserRepository.ts]: key is deletedAt not deleteAt.
Fix this key everywhere

File: tslint.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
   "rules": {
     "class-name": true,
     "comment-format": [true, "check-space"],
-    "indent": [true, "spaces", 2],
+    "indent": [false, "spaces", 2],
```
Comment (namanpar) [inline tslint.json]: keep indent true

**Review by namanpar — CHANGES_REQUESTED** [review]

File: tslint.json (hunk 5 above)
Comment (nx-mukesh) [reply tslint.json]: fixed

File: src/repositories/user/UserRepository.ts (hunk 4 above)
Comment (nx-mukesh) [reply src/repositories/user/UserRepository.ts]: fixed

File: src/libs/routes/authMiddleWare.ts (hunk 1 above)
Comment (nx-mukesh) [reply src/libs/routes/authMiddleWare.ts]: fixed

File: src/libs/routes/authMiddleWare.ts (hunk 2 above)
Comment (nx-mukesh) [reply src/libs/routes/authMiddleWare.ts]: fixed

**Review by namanpar — APPROVED** [review]

**Pull request merged**